pkg/plugin/plugin.go

```diff
@@ -1,21 +1,221 @@
+		return err
+	})
+}
+
+func copy(src, dst string) (int64, error) {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return 0, err
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return 0, fmt.Errorf("%s is not a regular file", src)
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return 0, err
+	}
+	defer source.Close()
+
+	destination, err := os.Create(dst)
+	if err != nil {
+		return 0, err
+	}
+	defer destination.Close()
+	nBytes, err := io.Copy(destination, source)
+	return nBytes, err
+}
+
+func createDownloadDir() (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	sourcePath := home + "/download_tls"
+	err = os.MkdirAll(sourcePath, 0755)
+	if err != nil {
+		return "", err
+	}
+
+	targetPath := filepath.Join(home, "/.lightning/testnet")
+
+	files := []string{"ca.pem", "client-key.pem", "client.pem"}
+
+	fmt.Println("")
+
+	for _, file := range files {
+		targetFile := filepath.Join(targetPath, file)
+		sourceFile := filepath.Join(sourcePath, file)
+		_, err := copy(targetFile, sourceFile)
+		if err != nil {
+			return "", err
+		}
+	}
+	return sourcePath, nil
+}
+
+type StartServer[T PluginState] struct{}
+
+func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]any) (map[string]any, error) {
+	if plugin.State.Server != nil {
+		return nil, fmt.Errorf("Server is already running")
+	}
+	gin.SetMode(gin.ReleaseMode)
+	router := gin.New()
+	router.GET("/ping", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{
+			"message": "pong",
+		})
+	})
+	router.POST("/tls", func(c *gin.Context) {
+		var login Login
+		if err := c.BindJSON(&login); err != nil {
+			c.JSON(404, gin.H{"error": err.Error()})
+			return
+		}
+		if plugin.State.Password != login.Password {
+			c.JSON(403, gin.H{"status": "Access Denied"})
+			return
+		}
+		targetPath, err := createDownloadDir()
+		if err != nil {
+			c.String(403, err.Error())
+			return
+		}
+		fileName := "certificates.zip"
+		zipTargetPath := filepath.Join(targetPath, fileName)
+		err = zipSource(targetPath, zipTargetPath)
+		if err != nil {
+			c.String(403, err.Error())
+			return
+		}
+		c.Header("Content-Description", "File Transfer")
+		c.Header("Content-Transfer-Encoding", "binary")
+		c.Header("Content-Disposition", "attachment; filename="+fileName)
+		c.Header("Content-Type", "application/octet-stream")
+		c.File(zipTargetPath)
+		os.RemoveAll(targetPath)
+	})
+	portValue, state := plugin.GetOpt("bender_port")
+
+	if state != true {
+		portValue = "8080"
+	}
+
+	plugin.State.Server = &http.Server{
+		Addr:    fmt.Sprintf(":%v", *&portValue),
+		Handler: router,
+	}
+
+	var serverError error
+
+	go func() {
+		// service connections
+		if err := plugin.State.Server.ListenAndServe(); err != nil {
+			// cannot panic or log here as this is in a goroutine
+			// throwing an to rpc call incase something comes up
+			serverError = fmt.Errorf("gin framework error: %s", err)
+		}
+		serverError = nil
```

Comment:
```suggestion
			serverError = fmt.Errorf("gin framework error: %s", err)
			return
		}
		serverError = nil
```

are you missing a return here? or you want to do this?